Use DateTime when the property is related to a Date

anthkernan-dev opened the pull request:
This commit will make date property use a Datetime instead of string.

# Pull Request

## What does this PR do?
Fixes #224 https://github.com/meilisearch/meilisearch-dotnet/issues/224

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to MeiliSearch!

anthkernan-dev committed:
Use DateTime when the property is related to a Date

This commit will make date property use a Datetime instead of string.

brunoocasali commented on tests/Meilisearch.Tests/KeyTests.cs:
```diff
@@ -105,7 +105,7 @@ public async Task DeleteOneKey()
                 Actions = new string[] { "documents.delete" },
                 Indexes = new string[] { "*" },
-                ExpiresAt = null,
+                ExpiresAt = DateTime.Parse("2042-04-02T00:42:42Z"),
```
I think this test doesn’t have to be changed, since the field support dates and null types, right?

anthkernan-dev replied:
Ah, I've reverted that test.

brunoocasali commented on tests/Meilisearch.Tests/KeyTests.cs:
```diff
@@ -29,7 +30,6 @@ public async Task GetAllKeys()
         {
             var keyResponse = await this.client.GetKeysAsync();
             var keys = keyResponse.Results;
-
```
Unintentional change? 🪀 

I like these spaces they help reading the structure of the test, do you know the **Four Phase-Test** structure? 
Knowing that I suggest you to revert this :D

Some refs:

- http://blog.plataformatec.com.br/2014/04/improve-your-test-readability-using-the-xunit-structure 
- http://xunitpatterns.com/Four%20Phase%20Test.html

anthkernan-dev replied:
I wasn't aware of the pattern, but I have read the links and I can understand why the space is there, I've also reverted.

brunoocasali reviewed (changes requested):
Hey @AK0785, thank you so much for your help here!

I left some comments for you, I'll approve after you address them 🥇 🍻

anthkernan-dev committed:
Fixed DeleteOneKey Test and Formatting

Fixed DeleteOneKey() test as the ExpiresAt field supports dates and null and also fixed test formatting.

brunoocasali reviewed (approved):
Thanks, @AK0785 for contributing with Meilisearch ❤️
